.github/workflows/primer_run_pr.yaml

```diff
@@ -0,0 +1,125 @@
+        run: >-
+          echo "::set-output name=key::venv-${{ env.CACHE_VERSION }}-${{
+            hashFiles('setup.cfg', 'requirements_test.txt', 'requirements_test_min.txt')
+          }}"
+      - name: Restore Python virtual environment
+        id: cache-venv
+        uses: actions/cache@v3.0.2
+        with:
+          path: venv
+          key: >-
+            ${{ runner.os }}-${{ steps.python.outputs.python-version }}-${{
+            steps.generate-python-key.outputs.key }}
+          restore-keys: |
+            ${{ runner.os }}-${{ steps.python.outputs.python-version }}-venv-${{ env.CACHE_VERSION }}-
+      - name: Create Python virtual environment
+        if: steps.cache-venv.outputs.cache-hit != 'true'
+        run: |
+          python -m venv venv
+          . venv/bin/activate
+          python -m pip install -U pip setuptools wheel
+          pip install -U -r requirements_test.txt
+
+      # Cache primer packages
+      - name: Download diffs
+        uses: actions/github-script@v6
+        with:
+          script: |
+            // Download 'main' pylint output
+            const fs = require('fs');
+            const { Octokit } = require("@octokit/rest");
+            const octokit = new Octokit({});
+            const runs = await octokit.rest.actions.listWorkflowRuns({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              workflow_id: ".github/workflows/primer_run_main.yaml",
+              status: "completed"
+            });
+            const lastRunMain = runs.data.workflow_runs.reduce(function(prev, current) {
+                return (prev.run_number > current.run_number) ? prev : current
+            })
+            console.log("Last run on main:")
+            console.log(lastRunMain.html_url)
+            const artifacts_main = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: lastRunMain.id,
+            });
+            const [matchArtifactMain] = artifacts_main.data.artifacts.filter((artifact) =>
+              artifact.name == "primer_commitstring");
+            const downloadWorkflow = await github.rest.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifactMain.id,
+               archive_format: "zip",
+            });
+            fs.writeFileSync("primer_commitstring.zip", Buffer.from(downloadWorkflow.data));
+      - run: unzip primer_commitstring.zip
+      - name: Get commit string
+        id: commitstring
+        run: |
+          . venv/bin/activate
+          output=$(python tests/primer/primer_tool.py prepare --read-commit-string)
+          echo "::set-output name=commitstring::$output"
+      - name: Restore projects cache
+        id: cache-projects
+        uses: actions/cache@v3
+        with:
+          path: .pylint_primer_tests/
```

Comment:
Should we add a .gitignore to this directory to ensure it always exists? I'm thinking for local testing etc.

Reply:
Isn't this a ``.gitkeep`` ?

Reply:
Added!